mongo-statefulset/config_server/template.yaml

```diff
@@ -0,0 +1,165 @@
+kind: Template
+apiVersion: v1
+metadata:
+  name: mongodb-configserver-aws-on-openshift
+  annotations:
+    openshift.io/display-name: "AWS MongoDB Config Server"
+    description: "MongoDB Config Server on OpenShift, AWS Hosted"
+    iconClass: "icon-mongodb"
+    tags: "database,mongodb,nosql,config"
+parameters:
+
+  - name: MONGODB_CONFIGSVR_REPLICA_NAME
+    displayName: "ConfigServer Replica Set Name"
+    description: "The name of the configuration server replica set."
+    value: cs0
+    required: true
+
+  - name: MONGODB_ADMIN_PASSWORD
+    displayName: "MongoDB Admin Password"
+    description: "Password for the admin user."
+    generate: expression
+    from: "[a-zA-Z0-9]{16}"
+    required: true
+
+  - name: MONGODB_KEYFILE_VALUE
+    displayName: "Keyfile Content"
+    description: "The value of the MongoDB keyfile for inter-cluster communication. Must be the same throughout the cluster. See (https://docs.mongodb.com/manual/core/security-internal-authentication/#internal-auth-keyfile)."
+    generate: expression
+    from: "[a-zA-Z0-9]{255}"
+    required: true
+
+  - name: MONGODB_IMAGE
+    displayName: "MongoDB Docker Image"
+    description: "A reference to a supported MongoDB Docker image."
+    value: "bornemannjs/mongodb-34:0.71"
+    required: true
+
+  - name: VOLUME_CAPACITY
+    displayName: "Volume Capacity"
+    description: "Volume space available for data, e.g. 512Mi, 2Gi."
+    value: "1Gi"
+    required: true
+
+  - name: MEMORY_LIMIT
+    displayName: "Memory Limit"
+    description: "Maximum amount of memory the container can use."
+    value: "512Mi"
+
+  - name: ADDITIONAL_STARTUP_OPTS
+    displayName: "Additional Startup Opts"
+    description: "Additional configuraiton options when starting the configuration server"
+    required: false
+    value: ""
+
+objects:
+  # A headless service to create DNS records
+  - kind: Service
+    apiVersion: v1
+    metadata:
+      name: "mongodb-configsvr-internal"
```

Comment:
Can we parameterize the names of the objects so they can be customized in some way? Generally something like ${APP_NAME}-configsvr-internal is the way to go.

Reply:
👍